pkg/file/builder.go

```diff
@@ -1033,6 +1037,65 @@ func (b *stateBuilder) rbacRoles() {
 	}
 }
 
+var (
+	IPv6HasPortPattern    = regexp.MustCompile(`\]\:\d+$`)
+	IPv6HasBracketPattern = regexp.MustCompile(`\[\S+\]$`)
+)
+
+func isIPv6(hostname string) bool {
+	parts := strings.Split(hostname, ":")
+	return len(parts) > 2
+}
+
+// expandIPv6 decompress an ipv6 address into its 'long' format.
+// for example:
+//
+// from ::1 to 0000:0000:0000:0000:0000:0000:0000:0001.
+func expandIPv6(address string) string {
+	ip := net.ParseIP(address).To16()
+	dst := make([]byte, hex.EncodedLen(len(ip)))
+	hex.Encode(dst, ip)
+	var final string
+	for i := 0; i < len(dst); i += 4 {
+		final += fmt.Sprintf("%s:", dst[i:i+4])
+	}
+	if len(final) == 0 {
+		return ""
+	}
+	// remove last colon
+	return final[:len(final)-1]
+}
+
+func normalizeIPv6(target string) (string, error) {
```

Comment:
This one doesn't have a godoc with an example of a UT so I'm not sure what it does but perhaps it can also be replaced with functions from `netip`?

Reply:
Added some description, let me know if that helps.